### PR TITLE
Add support for metrics aggregations to mvt end point

### LINKED ISSUE #77072: Support percentiles as an agg for _mvt format
Support the percentile aggregation for `_mvt`-requests. This aggregation is supported in Kibana Maps.

cc @iverase

### PR DESCRIPTION
**Pull request by iverase**
After adding in https://github.com/elastic/elasticsearch/pull/78594 a new abstraction for aggregation builders, we can exploit it in the vector tile implementation to add support for any of the aggregations that uses the abstraction. In particular this PR adds support for the following aggregations:

* `boxplot`
* `extended stats`
* `median absolute deviation`
* `percentile`
* `percentile-rank`
* `stats`
* `value count`

closes https://github.com/elastic/elasticsearch/issues/77072

### CODE REVIEW

**Review by jrodewig — APPROVED** [review]
Docs LGTM. Thanks @iverase!

File: x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java [hunk 1]
```diff
@@ -328,31 +324,25 @@ private void setSize(int size) {
...
-            switch (type) {
-                case MinAggregationBuilder.NAME:
-                case MaxAggregationBuilder.NAME:
-                case AvgAggregationBuilder.NAME:
-                case SumAggregationBuilder.NAME:
-                case CardinalityAggregationBuilder.NAME:
-                    break;
-                default:
-                    // top term and percentile should be supported
-                    throw new IllegalArgumentException("Unsupported aggregation of type [" + type + "]");
```
Comment (imotov) [inline x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java:347]: Nice!

File: x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java [hunk 2]
```diff
@@ -328,31 +324,25 @@ private void setSize(int size) {
...
+            if (aggregation instanceof MetricsAggregationBuilder<?, ?>) {
+                aggs.add((MetricsAggregationBuilder<?, ?>) aggregation);
+            } else {
+                throw new IllegalArgumentException("Unsupported aggregation of type [" + aggregation.getType() + "]");
```
Comment (imotov) [inline x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java:337]: Might be nice to clarify that only metrics aggregations are supported or something like this.

**Review by imotov — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**